### DISCUSSION TITLE
Support for remembering popouts on multiple monitors

### DISCUSSION THREAD

**Discussion by JonatanRydh** [caplin/FlexLayout#463] [Ideas]
Currently, when popout windows are restored they are confined to the current screen. However at least Chrome and Edge supports opening windows on other screens. This almost works with the current code since "off-screen" positions are saved. However when restoring windows there are currently two issues:

1. One needs to have asked the user for permission to manipulate multiple screens by accessing window.getScreenDetails(). Otherwise any opened window will be opened on the current screen.
2. Currently at least two functions `isOnScreen` and `keepOnScreen` is used to reposition windows to the current one.

1 is easy enough to do from the user of the library (we made a custom popout icon and show the permission when clicking on it, if getScreenDetails is supported and multiple monitors are detected by checking `window.screen.isExtended`) but for 2 we had to make a patch to make it work by overwriting Model.fromJson.

It would be nice if the library could handle both of these points to make popouts work properly on multiple screens by default - at least by the browsers that supports it.

(As a side-note: 2 also half-way works because `keepOnScreen` only checks if it is out of bounds greater than width/height and not if it is below 0. So popout panels on screens to the left of the main window actually already restores on that screen.)

**Comment by nealus** [comment]
both isOnScreen and keepOnScreen have been removed in v0.8.3, so that should fix part of your issue
